src/lib.rs

```diff
@@ -722,10 +738,20 @@ pub fn down(
 }
 
 /// Run the testing script.
-pub fn run(script: &Option<Script>) -> Result<(), Error> {
-    if let Some(ref code) = script {
+pub async fn run(config: &Config) -> Result<(), Error> {
+    if let Some(ref code) = config.run {
+        if !config.users.is_empty() {
+            for user in &config.users {
+                ensure_user_exists(
+                    &config.homeserver_config.public_baseurl,
+                    &config.homeserver_config.registration_shared_secret,
+                    user,
+                )
+                .await
+                .unwrap_or_else(|err| panic!("Could not setup user {}: {}", user.localname, err));
```

Comment:
Please raise errors instead of panicking.

Reply:
No, we have to panic here or nothing will happen and it will look like the scripts failed and not something critical to the setup